dfl-param-ids.rst

```diff
@@ -0,0 +1,27 @@
+.. SPDX-License-Identifier: GPL-2.0
+
+========================================
+Device Feature List (DFL) Parameter Ids
+========================================
+
+Parameter IDs within a version 1 feature are often private to a feature GUID.
+A few are interpreted by OPAE or other runtime code.
+
+The first feature of an AFU holds the AFU's GUID. Within that first feature,
+the following parameters IDs are interpreted by the system:
+
+.. list-table:: AFU First Feature Parameter IDs
+   :widths: 2 2 2
+   :header-rows: 1
+
+   * - Parameter Name
+     - ID
+     - Version
+
+   * - Child AFU GUID list
+     - 1
```

Comment:
Parameter ID value of 1 is currently the only generic id:
https://github.com/OFS/linux-dfl/blob/b85ea95d086471afb4ad062012a4d73cd328fa86/drivers/fpga/dfl.h#L114
I would still suggest 0x100 or 0x5.

Reply:
What does "generic" mean? I could add it to the table. Should I describe it as feature-specific?

We should pick a range that we reserve for future use in the AFU first feature. Should I pick 0x10-0x1f?

Reply:
I would say generic in this case is the opposite of feature-specific. For example the generic MSI_X parameter could be used for any feature connected to a PCIe bus.

I think what you're saying is Child AFU is an AFU feature-specific parameter. This seems correct, but feature-specific parameter ids don't really need to be reserved per se; they just need to be different than the generic parameter ids. We should then reserve a range of ids for "generic" parameters (1, 0xf000-0xffff?)

Reply:
I think we have a misunderstanding.

The problem with the AFU first feature is the GUID is the AFU's GUID. There's no specific GUID to use as a reference for understanding the parameter ID. In this table, I'm trying to describe the parameter IDs that will be interpreted by OPAE when viewing the AFU first feature. There is never an interrupt descriptor here.

I'm thinking that maybe the current choice of 1 is fine but that I just need to make it more clear that this table is specifically for the AFU's primary feature header.